vignettes/vcr.Rmd

```diff
@@ -1,191 +1,172 @@
-  %\VignetteIndexEntry{Introduction to vcr}
+  %\VignetteIndexEntry{Getting started with vcr}
   %\VignetteEngine{knitr::rmarkdown}
   %\VignetteEncoding{UTF-8}
+editor:
+  markdown:
+    wrap: sentence
 ---
 
 ```{r echo=FALSE}
 knitr::opts_chunk$set(
 	comment = "#>",
-	collapse = TRUE,
-	warning = FALSE,
-	message = FALSE,
-  eval = FALSE
+	collapse = TRUE
 )
 ```
 
-`vcr` is an R port of the Ruby gem [VCR](https://github.com/vcr/vcr) (i.e., a translation, there's no Ruby here :))
+{vcr} records and replays HTTP requests so you can test your API package with speed and confidence.
+It makes your tests independent of your internet connection (so they work on CRAN!) and because your tests get much much faster, you can write even more, increasing the coverage of your package.
+vcr works with {crul}, {httr}, and {httr2}.
+This vignette will introduce you to the basics of vcr for testing.
+To learn more about testing web requests in general, we recommend that you check out the [HTTP testing book](https://books.ropensci.org/http-testing/) which has a lot more documentation on all the packages involved.
+Once you're familiar with vcr for testing, you'll also find it easy to use in vignettes and examples: see `setup_knitr()` and `insert_example_cassette()` for the details.
 
-`vcr` helps you stub and record HTTP requests so you don't have to repeat HTTP requests.
+Here we'll use httr2 for generating the requests.
+The same principles apply if you're working with crul or httr, just the code for making requests will look different.
+I'm also going to start up a local httpbin server using webfakes.
+This will let us make some requests to a server that we're running locally so we don't need to worry about the internet being down.
 
-The main use case is for unit tests, but you can use it outside of the unit test use case.
-
-`vcr` works with the `crul`, `httr` and `httr2` HTTP request packages.
-
-Check out the [HTTP testing book](https://books.ropensci.org/http-testing/) for a lot more documentation on `vcr`, `webmockr`, and `crul`, and other packages.
-
-## Basic usage
-
-### In tests
+```{r setup}
+library(vcr)
+library(testthat)
+library(httr2, warn.conflicts = FALSE)
 
-In your tests, for whichever tests you want to use `vcr`, wrap them in a `vcr::use_cassette()` call like:
+httpbin <- webfakes::local_app_process(webfakes::httpbin_app())
+```
 
-```r
-library(testthat)
-vcr::use_cassette("rl_citation", {
-  test_that("my test", {
-    aa <- rl_citation()
-
-    expect_type(aa, "character")
-    expect_match(aa, "IUCN")
-    expect_match(aa, "www.iucnredlist.org")
-  })
-})
+```{r}
+#| include: false
+# Don't reocrd any cassette for this vignette so they're always created afresh
+vcr_configure(dir = tempdir())
 ```
 
-OR put the `vcr::use_cassette()` block on the inside, but put `testthat` expectations outside of
-the `vcr::use_cassette()` block:
+## Getting started
 
-```r
-library(testthat)
+The central metaphor of vcr is a video cassette.
```

Comment:
Should there be a link to explain this to young readers :joy:

Reply:
Do we also need a `rewind_casssette()` function that you need to call before returning the cassettes to the store?

Reply:
> Do we also need a rewind_casssette() function that you need to call before returning the cassettes to the store?

yes!